konnectivity-client/pkg/client/client.go

```diff
@@ -115,12 +115,13 @@ func (t *grpcTunnel) serve(c clientConn) {
 
 			if !ok {
 				klog.V(1).Infoln("DialResp not recognized; dropped")
+				return
```

Comment:
2 concerns but I think we should deal with them later.
1) Ideally we would send a PacketType_DIAL_CLS is c was a grpc channel. For that to happen c should be a *client.conn rather than a *grpc.ClientConn. (http-connect is a separate issue)
2) I would like to get away for SingleUseGRPCTunnel. I think this is wrong for that. However we can fix this change to work with that change when we are ready.

Reply:
Oh, will look into 1. I'm not sure if we close the connection here, if the proxy-server also needs to close some connection with the agent too. Should the DIAL_CLS solve that (if c was *client.conn)?

I'll have a look in the next days, but let me know if you already know the answer :)

Reply:
> Should the DIAL_CLS solve that (if c was *client.conn)?
Yes client.conn::Close will send either a CLOSE_REQ or DIAL_CLS depending on if it has received a DIAL_RSP. I think there is a bit of an issue in that we're crossing the streams. The 'grpc.clientConn' is the tunnel connection, in fact we later wrap it in a tunnel connection which makes that more clear. 'client.conn' is the request connection. So arguably what we should probably do is have tunnel Close() which walks the list of connections and calls Close on each of them.